metadata-ingestion/src/datahub/ingestion/source/mongodb.py

```diff
@@ -59,7 +72,7 @@
 DENY_DATABASE_LIST = set(["admin", "config", "local"])
 
 
-class MongoDBConfig(PlatformInstanceConfigMixin, EnvConfigMixin):
+class MongoDBConfig(DatasetSourceConfigMixin, StatefulIngestionConfigBase):
```

Comment:
While it's true that DatasetSourceConfigMixin just combines the other two
```
class DatasetSourceConfigMixin(PlatformInstanceConfigMixin, EnvConfigMixin):
```

we're generally trying to move away from tall config inheritance hierarchies, and would prefer wide hierarchies that are more explicit and understandable

that's a long way of saying that I'd prefer if it was 
```suggestion
class MongoDBConfig(PlatformInstanceConfigMixin, EnvConfigMixin, StatefulIngestionConfigBase):
```